catalog/CatalogRelationSchema.cpp

```diff
@@ -163,10 +166,16 @@ attribute_id CatalogRelationSchema::addAttribute(CatalogAttribute *new_attr) {
       ++num_nullable_attributes_;
       nullable_attribute_indices_.push_back(++current_nullable_attribute_index_);
     } else {
-      nullable_attribute_indices_.push_back(-1);
+      nullable_attribute_indices_.push_back(kInvalidCatalogId);
     }
 
-    return new_attr->getID();
+    const attribute_id attr_id = new_attr->getID();
+    // Ensure that we will not overrun the vector by resizing and initializing
+    // new entries to zero.
+    max_byte_lengths_.resize(attr_id + 1, 0);
```

Comment:
In `addAttribute`, we always append the new attribute to the end, although it may change in near future.